traffic_ops/traffic_ops_golang/asns.go

```diff
@@ -32,20 +32,23 @@ import (
 
 const ASNSPrivLevel = 10
 
-func ASNsHandler(db *sqlx.DB) AuthRegexHandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request, p PathParams, username string, privLevel int) {
+func ASNsHandler(db *sqlx.DB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Consider putting the `db` in the `Context` via a middleware func. I'm not a fan of `Context`, but if we're using it, might as well go all the way, and get the advantage of native funcs and `http.HandlerFunc`. 

If you do that, instead of ``` func ASNsHandler(db *sqlx.DB) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {```
Handlers become `func (w http.ResponseWriter, r *http.Request) {` and registration passes the function itself as an object, rather than calling the func and setting the handler to the object it returns. 

Seems easier to read and understand, especially for people who aren't FP Wizards.